llvm/test/CodeGen/AArch64/add.ll

```diff
@@ -232,7 +231,7 @@ define void @v2i16(ptr %p1, ptr %p2) {
 ; CHECK-SD-NEXT:    ld1 { v0.h }[2], [x8]
 ; CHECK-SD-NEXT:    ld1 { v1.h }[2], [x9]
 ; CHECK-SD-NEXT:    add v0.2s, v0.2s, v1.2s
-; CHECK-SD-NEXT:    mov s1, v0.s[1]
```

Comment:
Can such changes be avoided? Not sure if they'll be a performance impact but in general I'd rather not switch away from the output most people would expect to see.

Reply:
This didn't exist in the previous fold (which extracted an FPR), but is an artifact of inserting into a vector and then extracting the scalar FPR as a sub register. 

I'll see if it can be avoided while keeping the DAG combine generic enough to handle `i8` stores.

Reply:
I've added a couple of patterns to `Neon_INS_elt_pattern` to prefer the scalar `mov` format in this case.